test/IDE/complete_precedencegroup.swift

```diff
@@ -0,0 +1,117 @@
+  assignment: false
+  lowerThan: AdditionPrecedence
+  higherThan: #^PRECEDENCE_GROUP_3^#
+}
+precedencegroup MyPrecedence8 {
+  associativity: right
+  assignment: #^ASSIGNMENT_1^#
+}
+
+// Test that we have completions despite successive attributes being invalid
+precedencegroup MyPrecedence9 {
+  associativity: #^ASSOCIATIVITY_2^#
+  assignment: 5
+  higherTh:
+}
+precedencegroup MyPrecedence10 {
+  assignment: #^ASSIGNMENT_2^#
+  lower:
+  associativity: true
+}
+precedencegroup MyPrecedence11 {
+  higherThan: #^PRECEDENCE_GROUP_4^#
+  associativity: 3
+  lowerTh:
+}
+
+// Test that we see new precedence group declarations
+operator <><>: #^PRECEDENCE_GROUP_CURR_FILE^#
+
+
+// ATTRIBUTE_LIST: Begin completions, 4 items
+// ATTRIBUTE_LIST: Keyword/None: associativity; name=associativity
+// ATTRIBUTE_LIST: Keyword/None: higherThan;    name=higherThan
+// ATTRIBUTE_LIST: Keyword/None: lowerThan;     name=lowerThan
+// ATTRIBUTE_LIST: Keyword/None: assignment;    name=assignment
+
+// ASSOCIATIVITY: Begin completions, 3 items
+// ASSOCIATIVITY: Keyword/None: none;   name=none
+// ASSOCIATIVITY: Keyword/None: left;   name=left
+// ASSOCIATIVITY: Keyword/None: right;  name=right
+
+// ASSIGNMENT: Begin completions, 2 items
+// ASSIGNMENT: Keyword[false]/None: false; name=false
+// ASSIGNMENT: Keyword[true]/None: true;   name=true
+
+// PRECEDENCE_GROUP: Begin completions
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: BitwiseShiftPrecedence; name=BitwiseShiftPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: AssignmentPrecedence; name=AssignmentPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: ComparisonPrecedence; name=ComparisonPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: NilCoalescingPrecedence; name=NilCoalescingPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: RangeFormationPrecedence; name=RangeFormationPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: TernaryPrecedence; name=TernaryPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: DefaultPrecedence; name=DefaultPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: LogicalDisjunctionPrecedence; name=LogicalDisjunctionPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: LogicalConjunctionPrecedence; name=LogicalConjunctionPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: CastingPrecedence; name=CastingPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: AdditionPrecedence; name=AdditionPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: MultiplicationPrecedence; name=MultiplicationPrecedence
+// PRECEDENCE_GROUP: Decl[PrecedenceGroup]/OtherModule[Swift]: FunctionArrowPrecedence; name=FunctionArrowPrecedence
+
+// PRECEDENCE_GROUP_CURR_FILE-DAG: Decl[PrecedenceGroup]/CurrModule: MyPrecedence1; name=MyPrecedence1
```

Comment:
Why are the current file results separated?  Shouldn't you get all the precedence groups in the stdlib and current file every time?

And we should probably have a test for a precedence group in another source file.

Reply:
This is because I was having trouble getting precedence groups that are declared *after* the current one, they didn't show up. I thought this was happening because the rest hadn't been parsed yet and that's ok, but it doesn't really make sense now. I'll let you know if the issue is still present.

> And we should probably have a test for a precedence group in another source file.

It's a good idea, but I'll need a hint on how to write a multifile test. Do you have to manually create a module for that?

Reply:
> I was having trouble getting precedence groups that are declared after the current one, they simply didn't show up.

Okay, we should fix that.  If it turns out to be hard to fix I'm okay with taking this change without it though.

> I'll need a hint on how to write a multifile test. Do you have to manually create a module for that?

No need for another module, just another file in the current module.  You can add another file with a precedence group in the `test/IDE/Inputs` directory and then add that file to the command line invocation of swift-ide-test on the RUN line.  See `complete_multiple_files.swift` for an example of how to do that.